feat: new page for annotation processors (#5499)
JetBrains/kotlin-web-site: docs/topics/compiler-plugins/kapt.md

[//]: # (title: kapt compiler plugin)

<tldr>

* Use **kapt** if:
   * You have a Maven project.

>    * You have a Gradle project, but the required Java annotation processor does not yet support KSP. [See the list of supported libraries](ksp-overview.md#supported-libraries).

Reviewer: ```suggestion
   * You have a Gradle project, but the required Java annotation processor doesn't support KSP yet. [See the list of supported libraries](ksp-overview.md#supported-libraries).
```

* Use **[KSP](ksp-overview.md)** if:
   * You have a Gradle project, and the required Java annotation processor supports KSP.
   * You want to create your own annotation processors.

</tldr>

The kapt compiler plugin allows you to use existing Java annotation processors in Kotlin and works with both Maven and Gradle.
It generates stub files from Kotlin source code and then runs the Java annotation processors on those stubs.

> In a nutshell, kapt helps you use libraries like [MapStruct](https://mapstruct.org/) and 

Reviewer: In a nutshell reads a bit weird to me (I know this was in the old version).

What do you think about connecting it to the previous part like: 

This enables Java-based annotation processing in your Kotlin projects for libraries like [MapStruct](https://mapstruct.org/) and [Data Binding](https://developer.android.com/topic/libraries/data-binding/index.html).

```

> If you use Maven, you need to specify concrete plugin settings.

Reviewer: ```suggestion
If you use Maven, configure the plugin explicitly.
```

> See this [example of settings for the Lombok compiler plugin](lombok.md#using-with-kapt).

Reviewer: ```suggestion
See the following [example for the Lombok compiler plugin](lombok.md#using-with-kapt).
```
